sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -239,10 +331,10 @@ private[jdbc] class JDBCRDD(
   /**
    * `filters`, but as a WHERE clause suitable for injection into a SQL query.
    */
-  private val filterWhereClause: String =
-    filters
-      .flatMap(JDBCRDD.compileFilter(_, JdbcDialects.get(url)))
-      .map(p => s"($p)").mkString(" AND ")
+  private val filterWhereClause: String = {
+    val dialect = JdbcDialects.get(url)
+    predicates.flatMap(dialect.compileExpression(_)).map(p => s"($p)").mkString(" AND ")
```

Comment:
Since now `compileExpression` is used instead of `compileFilter` to construct the sql String for filters, then `compileFilter` is only used by V1 `JDBCRelation.unhandledFilters`. We probably should simplify and rename `compileFilter`.

Reply:
It seems I can remove `compileFilter` and implement `JDBCRelation.unhandledFilters` with `compileExpression`.